src/neo/SmartContract/Helper.cs

```diff
@@ -27,12 +28,13 @@ public static class Helper
             ApplicationEngine.OpCodePrices[OpCode.SYSCALL] +
             ApplicationEngine.ECDsaVerifyPrice * n;
 
-        public static UInt160 GetContractHash(UInt160 sender, byte[] script)
+        public static UInt160 GetContractHash(UInt160 sender, byte[] script, BigInteger nonce)
```

Comment:
I think we can compute the hash with `sender` and `checksum(nef)`.

Reply:
```
[ContractMethod(0, CallFlags.WriteStates | CallFlags.AllowNotify)]
private ContractState CreateChild(ApplicationEngine engine, string newName, StackItem data)
```

Reply:
If we use the manifest, we can avoid the new method, because it will be easy to change the name before deploy a new contract

Reply:
The nef checksum has already been computed.

https://github.com/neo-project/neo/blob/d9f7a3c1b853ac1a7799dc6c4069b91394aa9b3e/src/neo/SmartContract/NefFile.cs#L52

And it includes the script.

Reply:
changed from script to checksum(script)